common/ping.c

```diff
@@ -53,3 +59,25 @@ u8 *make_ping(const tal_t *ctx, u16 num_pong_bytes, u16 padlen)
 	tal_free(ignored);
 	return ping;
 }
+
+const char *got_pong(const u8 *pong, size_t *num_pings_outstanding)
+{
+	u8 *ignored;
+	int i;
+
+	if (!fromwire_pong(pong, pong, &ignored))
+		return "Bad pong";
+
+	if (*num_pings_outstanding == 0)
+		return "Unexpected pong";
+
+	for (i = 0; i < tal_len(ignored); i++) {
+		if (ignored[i] < ' ' || ignored[i] == 127)
```

Comment:
Maybe also check for `ignored[i] >= 128`?  The other side may dump random data in the pong, that by chance may have a sequence of >=128 bytes at the start, but do not form valid UTF-8 data.  Our pong here is just the version, which is composed solely of ASCII characters, so restricting to ASCII should be OK.

Reply:
In future we might stuff more information into the pong response; I'm not so so worried about malformed UTF-8 as I am confusion in the logs...